aloscene/scene_flow.py

```diff
@@ -0,0 +1,131 @@
+import aloscene
+from aloscene import Depth, CameraIntrinsic, Flow, Mask
+from typing import Union
+import numpy as np
+import torch
+
+
+def load_scene_flow(path: str) -> torch.Tensor:
+    with open(path, "rb") as file:
+        data = np.load(file)
+        return torch.from_numpy(data)
+
+
+def create_point_3d(depth: Depth, intrinsic: CameraIntrinsic) -> np.ndarray:
+    points3d = depth.as_points3d(camera_intrinsic=intrinsic).cpu().numpy()
+    mask_points = np.isfinite(points3d).all(1)
+    points3d = points3d[mask_points]
+    return points3d
+
+
+class SceneFlow(aloscene.tensors.SpatialAugmentedTensor):
+    """
+    Scene flow map
+
+    Parameters
+    ----------
+    x : str
+        load scene flow from a numpy file
+    """
+
+    @staticmethod
+    def __new__(cls, x, occlusion: Mask = None, *args, names=("C", "H", "W"), **kwargs):
+        if isinstance(x, str):
+            # load flow from path
+            x = load_scene_flow(x)
+            names = ("C", "H", "W")
+
+        tensor = super().__new__(cls, x, *args, names=names, **kwargs)
+        tensor.add_child("occlusion", occlusion, align_dim=["B", "T"], mergeable=True)
+        return tensor
+
+    def __init__(self, x, *args, **kwargs):
+        super().__init__(x)
+
+    @classmethod
+    def from_optical_flow(
+        cls,
+        optical_flow: Flow,
+        depth: Depth,
+        next_depth: Depth,
+        intrinsic: CameraIntrinsic,
+    ):
+        """Create scene flow from optical flow, depth a T, depth at T + 1 and the intrinsic
+
+        Parameters
+        ----------
+        optical flow: aloscene.Flow
+            The optical flow at T.
+        depth: aloscene.Depth
+            The depth at T."
+        next_depth: aloscene.Depth
+            The depth at T + 1
+        intrinsic : aloscene.CameraIntrinsic
+            The intrinsic of the image at T.
+        """
+        start_vector = np.reshape(depth.as_points3d(camera_intrinsic=intrinsic).cpu().numpy(), (depth.H, depth.W, 3))
+        next_vector = np.reshape(next_depth.as_points3d(camera_intrinsic=intrinsic).cpu().numpy(), (depth.H, depth.W, 3))
+
+        flow = np.round(optical_flow.numpy(), 0).astype(int)
+
+        y_coords, x_coords = np.mgrid[: depth.H, : depth.W]
+
+        # How to handle this
+        new_x = np.clip(x_coords + flow[:, :, 0], 0, depth.W - 1)
+        new_y = np.clip(y_coords + flow[:, :, 1], 0, depth.H - 1)
+
+        final_points = next_vector[new_y, new_x, :] - start_vector
```

Comment:
final_points is not a good names, because it's already a sceneflow vector (difference of two 3D points)